### PR TITLE
Support searching for enterprises from EnterpriseList

### PR DESCRIPTION
**Pull request by adamstankiewicz**
This PR adds support for searching for enterprise(s) from the `EnterpriseList` component, using Paragon's `SearchField`. 

On desktop, the search box is positioned to the right; on mobile, it drops below the "Enterprise List" heading and expands to full width.

image

### CODE REVIEW

File: src/components/Admin/Admin.scss [hunk 1]
```diff
@@ -10,7 +10,3 @@
         flex: 1;
     }
 }
-
-.download-btn div {
```
Comment (adamstankiewicz) [inline src/components/Admin/Admin.scss]: With the most recent Paragon release, the `div` in the `Icon` component was changed to a `span` so this CSS is no longer needed.

File: src/components/EnterpriseList/index.jsx [hunk 2]
```diff
@@ -126,23 +173,33 @@ class EnterpriseList extends React.Component {
...
+          </div>
+          <div className="row mt-2">
+            <div className="col">
               {error && this.renderErrorMessage()}
               {loading && !enterprises && this.renderLoadingMessage()}
               {!loading && !error && enterprises && enterprises.length === 0 &&
                 this.renderEmptyEnterpriseListMessage()
               }
-              {!loading && !error && enterprises && enterprises.length === 1 && pageCount === 1 &&
+              {this.shouldRenderRedirectToEnterpriseAdminPage() &&
```
Comment (adamstankiewicz) [inline src/components/EnterpriseList/index.jsx]: Moving this conditional to its own function to reduce some complexity in the `render` function.

File: src/index.scss [hunk 3]
```diff
@@ -3,3 +3,6 @@
 
 $fa-font-path: "~font-awesome/fonts";
 @import "~font-awesome/scss/font-awesome";
+
+@import "~@edx/paragon/src/SearchField/SearchField";
```
Comment (adamstankiewicz) [inline src/index.scss]: To make Paragon styles work properly, we need to include the `scss` file for any component that has non-native Bootstrap here.

File: src/components/EnterpriseList/EnterpriseList.test.jsx [hunk 4]
```diff
@@ -159,11 +178,55 @@ describe('<EnterpriseList />', () => {
...
+
+      // onSubmit
+      wrapper.find('SearchBar').find('input[type=\'search\']').simulate('change', { target: { value: searchQuery } });
+      expect(wrapper.find('SearchBar').find('input[type=\'search\']').prop('value')).toEqual(searchQuery);
+      wrapper.find('SearchBar').find('.input-group-append').find('button').last()
+        .simulate('click');
+      expect(wrapper.find('EnterpriseList').instance().state.searchQuery).toEqual(searchQuery);
+      expect(wrapper.find('EnterpriseList').instance().state.searchSubmitted).toBeTruthy();
+
+      // // onClear
```
Comment (georgebabey) [inline src/components/EnterpriseList/EnterpriseList.test.jsx]: nit: double `//`s

File: src/components/EnterpriseList/EnterpriseList.test.jsx [hunk 5]
```diff
@@ -159,11 +178,55 @@ describe('<EnterpriseList />', () => {
...
         enterprises={null}
       />
-    ));
-    expect(wrapper.find('EnterpriseList').instance().state.pageCount).toEqual(null);
+    )).find('EnterpriseList');
+    expect(wrapper.dive().state('pageCount')).toEqual(null);
+  });
+
+  describe('enterprise list search', () => {
+    it('searchQuery state changes onSearch and onClear', () => {
```
Comment (georgebabey) [inline src/components/EnterpriseList/EnterpriseList.test.jsx]: nit: I'd break this into 2 tests

**Pull request merged**